src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResourceIntTest.kt

```diff
@@ -161,6 +176,97 @@ class ActivitiesResourceIntTest : IntegrationTestBase() {
     }
   }
 
+  @Nested
+  inner class UpdateActivity {
+
+    private fun updateActivityRequest() = UpdateActivityRequest(prisonId = "LEI")
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.put().uri("/activities/1")
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(BodyInserters.fromValue(updateActivityRequest()))
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.put().uri("/activities/1")
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(roles = listOf()))
+        .body(BodyInserters.fromValue(updateActivityRequest()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden with wrong role`() {
+      webTestClient.put().uri("/activities/1")
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+        .body(BodyInserters.fromValue(updateActivityRequest()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `should call the service`() {
+      doReturn(UpdateActivityResponse(prisonId = "LEI")).whenever(activityService).updateActivity(anyLong(), any())
+
+      webTestClient.put().uri("/activities/1")
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ACTIVITIES")))
+        .body(
+          BodyInserters.fromValue(
+            """{
+            "prisonId" : "$prisonId"
+          }"""
+          )
+        )
+        .exchange()
+        .expectStatus().isOk
+
+      verify(activityService).updateActivity(1, updateActivityRequest())
+    }
+
+    @Test
+    fun `should return bad request`() {
+      doThrow(BadDataException("Prison not found")).whenever(activityService).updateActivity(anyLong(), any())
```

Comment:
similar comment to above, do you need a spy bean - wouldn't a missing prison ID give you the same result?